code/modules/hippie/pool/pool_controller.dm

```diff
@@ -0,0 +1,363 @@
+
+//Originally stolen from paradise. Credits to tigercat2000.
+//Modified a lot by Kokojo.
+/obj/machinery/poolcontroller
+	name = "Pool Controller"
+	desc = "A controller for the nearby pool."
+	icon = 'icons/turf/pool.dmi'
+	icon_state = "poolcnorm"
+	anchored = 1
+	density = 1
+	use_power = 1
+	idle_power_usage = 75
+	var/list/linkedturfs = list() //List contains all of the linked pool turfs to this controller, assignment happens on New()
+	var/temperature = "normal" //The temperature of the pool, starts off on normal, which has no effects.
+	var/srange = 6 //The range of the search for pool turfs, change this for bigger or smaller pools.
+	var/linkedmist = list() //Used to keep track of created mist
+	var/misted = 0 //Used to check for mist.
+	var/obj/item/weapon/reagent_containers/beaker = null
+	var/cur_reagent = "water"
+	//var/datum/wires/poolcontroller/wires = null
+	var/drainable = 0
+	var/drained = 0
+	var/bloody = 0
+	var/lastbloody = 99
+	var/obj/machinery/drain/linkeddrain = null
+	var/timer = 0 //we need a cooldown on that shit.
+	var/reagenttimer = 0 //We need 2.
+	var/shocked = 0//Shocks morons, like an airlock.
+
+/obj/machinery/poolcontroller/New() //This proc automatically happens on world start
+	wires = new /datum/wires/poolcontroller(src)
+	for(var/turf/open/pool/water/W in range(srange,src)) //Search for /turf/open/beach/water in the range of var/srange
+		src.linkedturfs += W
+	for(var/obj/machinery/drain/pooldrain in range(srange,src))
+		src.linkeddrain += pooldrain
+	..() //Always call your parents when you're a new thing.
+
+/obj/machinery/poolcontroller/emag_act(user as mob) //Emag_act, this is called when it is hit with a cryptographic sequencer.
+	if(!emagged) //If it is not already emagged, emag it.
+		user << "\red You disable \the [src]'s temperature safeguards." //Inform the mob of what emagging does.
+		emagged = 1 //Set the emag var to true.
+
+/obj/machinery/poolcontroller/attackby(obj/item/weapon/W, mob/user)
+	..()
+	if(issilicon(user))
+		return
+	if(istype(W, /obj/item/weapon/screwdriver) && anchored)
+		panel_open = !panel_open
+		user << "You [panel_open ? "open" : "close"] the maintenance panel."
+		overlays.Cut()
+		if(panel_open)
+			overlays += image(icon, "wires")
+		updateUsrDialog()
+		return
+	else if(istype(W, /obj/item/device/multitool)||istype(W, /obj/item/weapon/wirecutters))
+		if(panel_open)
+			attack_hand(user)
+		updateUsrDialog()
+		return
+
+	else
+		if(stat & (NOPOWER|BROKEN))
+			return
+		if (istype(W,/obj/item/weapon/reagent_containers/glass/beaker/large))
+			if(beaker)
+				user << "A beaker is already loaded into the machine."
+				return
+
+			if(W.reagents.total_volume >= 100 && W.reagents.reagent_list.len == 1) //check if full and allow one reageant only.
```

Comment:
This might be somewhat more of a feature request, but why don't we turn the pool into a proper reagent pool you can actually dump shit into? The current implementation is rather snowflakey